community/kernel/src/main/java/org/neo4j/kernel/impl/proc/Procedures.java

```diff
@@ -136,10 +136,22 @@ public void registerProcedure( Class<?> proc ) throws KernelException
     /**
      * Register a new procedure defined with annotations on a java class.
      * @param proc the procedure class
+     * @param overrideCurrentImplementation set to true if procedures within this class should override older procedures with the same name
      */
     public void registerProcedure( Class<?> proc, boolean overrideCurrentImplementation ) throws KernelException
     {
-        for ( CallableProcedure procedure : compiler.compileProcedure( proc ) )
+        registerProcedure( proc, overrideCurrentImplementation, false );
+    }
+
+    /**
+     * Register a new procedure defined with annotations on a java class.
+     * @param proc the procedure class
+     * @param overrideCurrentImplementation set to true if procedures within this class should override older procedures with the same name
+     * @param warn the warning the procedure should generate when called
```

Comment:
How can a boolean be "the warning the procedure should generate when called"? Maybe the javadocs was made and then the code changed? Really nice otherwise that you added a javadoc =).